### PR TITLE
[WIP] Get chip-tool to broadcast commissioner mdns advertisement

### LINKED ISSUE #15721: [chip-tool] Get `chip-tool` to broadcast commissioner advertising
#### Problem

For the discovery tests, a tool is needed to broadcast commissioner advertisement. It will be a nice addition to `chip-tool` but a simple way would be to just use one of the accessory with the `CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY` flag set to true
That said the previous workaround would not allow to inject custom txt entries, so dedicated command in `chip-tool` would be useful.

 #### Proposed Solution
 Add support via a dedicated command for that to `chip-tool`

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

Some tests needs to be runned against a commissioner with a custom txt field.

#### Change overview
 * Add a new `advertise-commissioner` command to `chip-tool` with optional parameters such as `interface-id` and `custom-txt-record`
 * Update both the platform mdns implementation and `minimal`
 
#### Testing
It was locally tested using both implementations and looking at the output with the `dns-sd` command.

fix #15721

### CODE REVIEW

File: src/app/server/Dnssd.h [hunk 1]
```diff
@@ -59,6 +59,15 @@ class DLL_EXPORT DnssdServer
     /// Gets the interface id used for advertising
     Inet::InterfaceId GetInterfaceId() { return mInterfaceId; }
 
+#ifdef CHIP_CONFIG_TEST
+    /// Sets a custom value for the unsupported txt record
+    /// This function does not copy the string until StartServer is called.
+    void SetUnsupportedTxtRecord(const char * value) { mUnsupportedTxtRecord = value; }
+
+    /// Gets the interface id used for advertising
```
Comment (bzbarsky-apple) [inline src/app/server/Dnssd.h:67]: This is not about interface ids.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
Fast tracking given this has had enough time for review.

**Comment by woody-apple** [comment]
@vivien-apple ?

**Pull request closed without merging**